File: solutions/HJK9810/2026-09-03/medium/1230.py
codes = []

def insert(orders, offset):
    global codes
    x, y = int(orders[offset]), int(orders[offset + 1])

    for idx in range(y):
        codes.insert(x + idx, orders[offset + 2 + idx])

    return offset + 2 + y

def add(orders, offset):
    global codes
    y = int(orders[offset])

    for idx in range(y):
        codes.append(orders[offset + 1 + idx])

    return offset + 1 + y

def delete(orders, offset):
    global codes
    x, y = int(orders[offset]), int(orders[offset + 1])
    for _ in range(y):
        codes.pop(x)

    return offset + 2

def algorithm():
    global codes

    N = int(input())
    codes = list(input().split())
    M = int(input())
    orders = list(input().split())

    offset = 0
    for _ in range(M):
        if orders[offset] == 'I':
            offset = insert(orders, offset + 1)
        elif orders[offset] == 'D':
            offset = delete(orders, offset + 1)
        elif orders[offset] == 'A':
            offset = add(orders, offset + 1)

    return " ".join(codes[0:10])

# T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, 11):
    print(f"#{test_case} {algorithm()}")

File: solutions/HJK9810/2026-09-03/medium/1230.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.StringTokenizer;

class Solution {
  private ArrayList<String> codes;
  private String[] orders;

  private int insert(int offset) {
    int x = Integer.parseInt(orders[offset]);
    int y = Integer.parseInt(orders[offset + 1]);

    for (int idx = 0; idx < y; idx++) {
      codes.add(x + idx, orders[offset + 2 + idx]);
    }

    return offset + 2 + y;
  }

  private int add(int offset) {
    int y = Integer.parseInt(orders[offset]);
    for (int idx = 0; idx < y; idx++) {
      codes.add(orders[offset + 1 + idx]);
    }

    return offset + 1 + y;
  }

  private int delete(int offset) {
    int x = Integer.parseInt(orders[offset]);
    int y = Integer.parseInt(orders[offset + 1]);

    for (int idx = 0; idx < y; idx++) {
      codes.remove(x);
    }

    return offset + 2;
  }

	public static void main(String args[]) throws Exception {
    Solution sol = new Solution();
    StringTokenizer st;
		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
    // int T = Integer.parseInt(input.readLine());
    StringBuilder sb = new StringBuilder();

		for(int test_case = 1; test_case <= 10; test_case++) {
      int N = Integer.parseInt(input.readLine());
      sol.codes = new ArrayList<>();
      st = new StringTokenizer(input.readLine());
      for (int idx = 0; idx < N; idx++) {
        sol.codes.add(st.nextToken());
      }

      int M = Integer.parseInt(input.readLine());
      sol.orders = input.readLine().split(" ");
      
      int offset = 0;
      for (int count = 0; count < M; count++) {
        switch (sol.orders[offset]) {
          case "I":
            offset = sol.insert(offset + 1);
            break;
          case "D":
            offset = sol.delete(offset + 1);
            break;
          case "A":
          default:
            offset = sol.add(offset + 1);
            break;
        }
      }

      sb.append("#").append(test_case).append(" ");
      for (int idx = 0; idx < 10; idx++) {
        sb.append(sol.codes.get(idx)).append(" ");
      }
      sb.append("\n");
		}
    System.out.print(sb.toString());
	}
}
